File: o2o_erpnext/o2o_erpnext/report/profit_and_loss_dashborad/profit_and_loss_dashborad.js
// Copyright (c) 2025, Ascratech LLP and contributors
// For license information, please see license.txt

frappe.query_reports["Profit and Loss Dashborad"] = {
	"filters": [

	]
};

// formatter: function (value, row, column, data, default_formatter) {
// 	value = default_formatter(value, row, column, data);

// 	if (column.fieldname === "name" && data && data.name) {
// 		// Link to open Purchase Order in the same tab:
// 		return `<a href="/app/purchase-order/${data.name}">${value}</a>`;

// 		// OR, if you want it in a new tab:
// 		// return `<a href="/app/purchase-order/${data.name}" target="_blank">${value}</a>`;
// 	}
// 	if (column.fieldname === "supplier" && data && data.supplier) {
// 		return `<a href="/app/supplier/${data.supplier}">${value}</a>`;
// 	}

// 	return value;
// }

File: o2o_erpnext/o2o_erpnext/report/profit_and_loss_dashborad/profit_and_loss_dashborad.py
# Copyright (c) 2025, Ascratech LLP and contributors
# For license information, please see license.txt

import frappe
from frappe import _

def execute(filters=None):
	columns = get_columns()
	data = get_data(filter)
	return columns, data

def get_data(fillter):
	data=frappe.db.sql('''select name,supplier,custom_sub_branch,transaction_date,custom__approved_at,custom_branch,custom_total_pl,custom_total_purchase_cost,total from `tabPurchase Order` ''', as_dict=True)
	return data

def get_columns():
	columns = [
		{
            "label": _("Supplier"),
            "fieldname": "supplier",
            "fieldtype": "data",
            "width": 150,
        },
		{
            "label": _("Sub Branch"),
            "fieldname": "custom_sub_branch",
            "fieldtype": "data",
            "width": 150,
        },
        {
            "label": _("Purchase Order"),
            "fieldname": "name",
            "fieldtype": "data",
            "width": 150,
        },
		{
            "label": _("Created At"),
            "fieldname": "transaction_date",
            "fieldtype": "data",
            "width": 150,
        },
		{
            "label": _("Approved At"),
            "fieldname": "custom__approved_at",
            "fieldtype": "data",
            "width": 150,
        },
		{
            "label": _("Total"),
            "fieldname": "total",
            "fieldtype": "data",
            "width": 100,
        },
		{
            "label": _("Total Purchase cost"),
            "fieldname": "custom_total_purchase_cost",
            "fieldtype": "data",
            "width": 100,
        },
		{
            "label": _("Total P&L"),
            "fieldname": "custom_total_pl",
            "fieldtype": "data",
            "width": 100,
        },
		{
            "label": _("Branch"),
            "fieldname": "custom_branch",
            "fieldtype": "data",
            "width": 100,
        },
		
	]
	return columns

